SmartPlaces.Facilities/product/Topology/product/src/Program.cs

```diff
@@ -0,0 +1,94 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="Program.cs" company="Microsoft">
+//   Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Topology
+{
+    using System;
+    using System.Threading.Tasks;
+    using Azure.Identity;
+    using Microsoft.Extensions.Options;
+    using IngestionManager;
+    using IngestionManager.Interfaces;
+    using IngestionManager.Mapped.Extensions;
+    using OntologyMapper;
+    using OntologyMapper.Mapped;
+
+    public class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            Console.WriteLine("Starting Topology Processing");
+
+            using IHost host = Host
+                .CreateDefaultBuilder(args)
+                .ConfigureAppConfiguration((hostContext, config) =>
+                {
+                    var settings = config.Build();
+
+                    var vaultUri = settings["KeyVaultEndpoint"];
+
+                    if (string.IsNullOrWhiteSpace(vaultUri))
+                    {
+                        Console.WriteLine("Startup Variable 'KeyVaultEndpoint' not found. Exiting");
+                        throw new ArgumentNullException("KeyVaultEndpoint");
+                    }
+
+                    config.AddAzureKeyVault(new Uri(vaultUri), new DefaultAzureCredential());
+                })
+                .ConfigureServices((hostContext, services) =>
+                {
+                    services.AddLogging();
+
+                    services.AddApplicationInsightsTelemetryWorkerService(options =>
+                    {
+                        options.ConnectionString = hostContext.Configuration["AppInsightsConnectionString"];
+                    });
+
+                    // Implements IInputGraphManager, IGraphIngestionProcessor, IOutputGraphManager, ITelemetryIngestionProcessor
+                    services.AddMappedIngestionManager(options =>
+                    {
+                        // Mapped Specific
+                        options.MappedToken = hostContext.Configuration["MappedToken"];
+                        options.MappedRootUrl = hostContext.Configuration["MappedRootUrl"];
+
+                        // Ingestion Manager
+                        options.AzureDigitalTwinsEndpoint = hostContext.Configuration["AzureDigitalTwinsEndpoint"];
+                    });
+
+                    // Ties Topology and Telemetry together
+                    // Implements IDistributedCache
+                    services.AddStackExchangeRedisCache(options =>
+                    {
+                        options.Configuration = hostContext.Configuration["RedisCacheConnectionString"];
+                    });
+                    services.AddSingleton<ITwinMappingIndexer, RedisTwinMappingIndexer>();
+
+                    // From opendigitaltwins-tools
```

Comment:
This comment and the next one dont make sense anymore as its all from opendigitaltwins-tools.